camel/agents/react_agent.py

```diff
@@ -0,0 +1,333 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ========= Copyright 2023-2024 @ CAMEL-AI.org. All Rights Reserved. =========
+
+from enum import Enum
+from typing import Any, Callable, Dict, List, Optional, Union
+
+from pydantic import BaseModel, Field
+
+from camel.agents import ChatAgent
+from camel.logger import get_logger
+from camel.messages import BaseMessage
+from camel.models import BaseModelBackend
+from camel.responses import ChatAgentResponse
+from camel.toolkits import FunctionTool
+from camel.types import RoleType
+from camel.utils import track_agent
+
+logger = get_logger(__name__)
+
+
+class ReActStep(BaseModel):
+    """Structured format for ReAct steps"""
+
+    thought: str = Field(description="Reasoning about current situation")
+    action: str = Field(description="Action to take (Search/Lookup/Finish)")
+    observation: Optional[str] = Field(
+        None, description="Results of the action"
+    )
+
+
+class ReActActionSpace(Enum):
+    r"""Available actions in the ReAct framework as
+    defined in the original paper.
+
+    References:
+        https://arxiv.org/pdf/2210.03629
+    """
+
+    SEARCH = "Search"
+    LOOKUP = "Lookup"
+    FINISH = "Finish"
+
+
+@track_agent(name="ReActAgent")
+class ReActAgent(ChatAgent):
+    r"""ReAct Agent that combines reasoning and acting through:
+    - Thought: Reasoning about current state
+    - Action: Deciding what action to take
+    - Observation: Getting results of actions
+
+    Args:
+        system_message (BaseMessage): The system message for initializing the
+            agent's conversation context.
+        model (Optional[BaseModelBackend], optional): The model backend to use
+            for response generation. (default: :obj:`None`)
+        tools (Optional[List[Union[FunctionTool, Callable]]], optional): List
+            of available tools that can be used to execute actions. Tools can
+            be either FunctionTool instances or callable functions.
+            (default: :obj:`None`)
+        max_steps (int, optional): Maximum number of reasoning steps before
+            forced termination. Prevents infinite loops.
+            (default: :obj:`10`)
+    """
+
+    def __init__(
+        self,
+        system_message: BaseMessage,
+        model: Optional[BaseModelBackend] = None,
+        tools: Optional[List[Union[FunctionTool, Callable]]] = None,
+        max_steps: int = 10,
+    ) -> None:
+        super().__init__(system_message=system_message, model=model)
```

Comment:
```suggestion
        super().__init__(system_message=system_message, model=model, tools=tools)
```
Camel's function can be exctuted automaticly by agent, Is this the functionality you need?